custom_components/elasticsearch/es_doc_publisher.py

```diff
@@ -164,14 +175,19 @@ def queue_size(self):
         """Return the approximate queue size."""
         return self.publish_queue.qsize()
 
-    def enqueue_state(self, state: State, event: EventType):
+    def enqueue_state(self, state: State, event: EventType, reason: str):
         """Queue up the provided state change."""
 
         domain = state.domain
         entity_id = state.entity_id
 
         if self._should_publish_entity_state(domain, entity_id):
-            self.publish_queue.put((state, event))
+            self.publish_queue.put((state, event, reason))
+
+    def empty_queue(self):
+        """Empty the publish queue."""
+        while not self.publish_queue.empty():
+            self.publish_queue.get()
```

Comment:
We can probably just initialize a new queue and let the runtime perform garbage collection.

```suggestion
        self.publish_queue = Queue[tuple[State, EventType, str]]()
```